docs: data studio updates (#72641)
metabase/metabase: docs/data-studio/measures.md

---
title: Measures
summary: Create saved aggregations on tables so people can use consistent calculations when building queries.
---

# Measures

Measures are saved aggregations associated with a table. You can use measures to create an official definition of a calculation—like Net Promoter Score—so everyone in your Metabase uses the same formula instead of writing their own versions.

People will see measures as options in the Summarize block of the [query builder](../questions/query-builder/editor.md).

![Measure in the query builder](./images/measure.png)

## Create measures

> > If your instance is in read-only [remote sync](../installation-and-operation/remote-sync.md) mode, and you enabled sync of the [Library](./library.md), you will not be able to create measures on tables published in the Library while in read-only mode.

Reviewer: This might be excessive. Do we have to say this for everything?

Author: We get so many people wondering why such and such action is unavailable, and the answer is always remote sync. Until the UI handles this gracefully, we should say it upfront.

Measures are created in [Data Studio](../data-studio/overview.md).

1. Go to **Data Studio** by clicking the **grid icon** in top right of the screen and selecting **Data Studio**.
2. In Data Studio, go to **Tables** in the left sidebar, and select the table to define a measure on.
3. In the right sidebar for the table, select **Measures**.

   ![Find measures](./images/find-measures.png)

4. Click **+New measure**. You'll see an abridged version of Metabase's query builder that focuses on aggregation.

5. Add your aggregation formula. For more complex aggregations, check out [custom expressions](../questions/query-builder/expressions.md).

   For example, you can define a measure like `(CountIf([Score] >= 9) - CountIf([Score] <= 6)) / Count() * 100` to calculate Net Promoter Score.

   ![Measure definition](./images/measure-definition.png)

6. To preview the results of the measure, click on the **three dots** icon next to the measure's name and select **Preview**.
7. Save.

## Use measures in the query builder

> Measures allow people to use pre-defined aggregations to summarize their data.
> 
> To use a measure in the query builder:

Reviewer: ```suggestion
```

1. Start a new question from the table that the measure is based on.
2. Select the measure in the **Summarize** block.

> The measure's saved aggregation will be applied behind the scenes. You can use breakouts with a saved measure.

Reviewer: ```suggestion
```

![Measure in the query builder](./images/measure.png)

> Measures will only appear in questions that use the measure's table as the primary data source. They will not appear in questions joining the measure's table, or in questions built on other questions or models, even if those questions or models use the measure's table as a source themselves.

Reviewer: ```suggestion
Measures are only available to questions that directly query the measure's table. Measures aren't available for questions that join the measure's table, or that involve nested queries (even if those queries measure's table as a source themselves).
```

## Delete measures

> Deleting a measure will not break questions using it. The questions that use the measure will keep using the same aggregations as before.

Reviewer: ```suggestion
Deleting a measure won't break questions using it. The questions that use the deleted measure will "revert" to using the measure's (now unnamed) aggregation formula.
```

1. Go to **Data Studio** by clicking the **grid icon** in top right of the screen and selecting **Data Studio**.
2. In Data Studio, go to **Tables**, and select the measure's table.
3. In the right sidebar for the table, select **Measures**.
4. Choose the measure you want to delete.
5. On the measure's page, click on the **three dots** icon next to the measure's name and select **Remove measure**.

> ## Explore measures
> 
> You can explore measures along dimensions and compare several measures in the [Metrics Explorer](../questions/metrics-explorer.md).
> 
> ![Explore a measure](./images/explore-measure.png)
> 
> To see all measures on a table, select the table in [Data Studio > Managing tables](./managing-tables.md) and switch to the **Measures** tab.

Reviewer: ```suggestion
```

Don't think we need this section. Maybe just refer to the explorer somewhere and link to it.

> ## Permissions for measures

Reviewer: I think this section is okay for now, but we may want to consolidate data studio perms in permissions somewhere so the info is less atomized.

To **create, edit, or delete** measures, people need to:

- Be admins or members of the [Data Analysts](../people-and-groups/managing.md#data-analysts) group. This is needed to access Data Studio.
- Have [Create queries](../permissions/data.md#create-queries-permissions) permissions on the measure's table. This is needed to create the measure's query.

To **use** a measure, people need to:

- Have [Create queries](../permissions/data.md#create-queries-permissions) permissions on the measure's table.

> ## Only measures on published tables are synced to Git

Reviewer: Just noting another thing we could consolidate (with remote sync docs). No action on this PR.
